p2p/discover/v5wire/encoding.go

```diff
@@ -278,7 +278,11 @@ func (c *Codec) encodeRandom(toID enode.ID) (Header, []byte, error) {
 	head.AuthData = c.headbuf.Bytes()
 
 	// Fill message ciphertext buffer with random bytes.
-	c.msgctbuf = append(c.msgctbuf[:0], make([]byte, randomPacketMsgSize)...)
+	if cap(c.msgctbuf) < randomPacketMsgSize {
+		c.msgctbuf = make([]byte, randomPacketMsgSize)
+	} else {
+		c.msgctbuf = c.msgctbuf[:randomPacketMsgSize]
+	}
```

Comment:
```suggestion
    c.msgctbuf = slices.Grow(c.msgctbuf[:0], randomPacketMsgSize)
```
Probably the same thing, see my other comment.